File: Sources/echoscrape/Utilities/Cache.swift
actor Cache {
    private var cache: [String: Any] = [:]

    func get<T>(key: String) -> T? {
        guard let item = cache[key] else {
            Logger.fail("(cache) Cache miss for '\(key)'...")
            return nil
        }

        Logger.success("(cache) Cache hit for '\(key)'. Returning cached data.")
        return item as? T
    }

    func set(key: String, val: Any) {
        self.cache[key] = val
        Logger.success("(cache) Data cached for '\(key)'.")
    }
}

File: lib/cache/cache.go
package cache

import (
	"echoscrape/lib/logger"
	"fmt"
)

type Cache struct {
	cacheMap map[string]any
}

func New() *Cache {
	c := &Cache{make(map[string]any)}
	return c
}

func (c *Cache) Get(key string) (any, bool) {
	item, exists := c.cacheMap[key]

	if !exists {
		logger.Fail(fmt.Sprintf("(cache) Cache miss for '%s'...", key))
		return nil, false
	}

	logger.Success(fmt.Sprintf("(cache) Cache hit for '%s'. Returning cached data.", key))

	return item, true
}

func (c *Cache) Set(key string, val any) {
	c.cacheMap[key] = val
	logger.Success(fmt.Sprintf("(cache) Data cached for '%s'.", key))
}
